src/Data/Symbol.purs

```diff
@@ -5,20 +5,30 @@ module Data.Symbol
   , SProxy(..)
   ) where
 
+import Type.Proxy (Proxy(..))
+
 -- | A value-level proxy for a type-level symbol.
-data SProxy (sym :: Symbol) = SProxy
+-- | **Deprecated as of v0.14.0 PureScript release**: use `Type.Proxy` instead.
+data SProxy :: Symbol -> Type
+data SProxy sym = SProxy
 
 -- | A class for known symbols
 class IsSymbol (sym :: Symbol) where
-  reflectSymbol :: SProxy sym -> String
+  -- Note: Before v0.14.0, we did not have polykinds. Thus, we needed
+  -- kind-specific proxy types to pass a Symbol around.
+  -- Once v0.14.0 was released, we could use the kind-generic `Proxy` type
+  -- instead. However, to reduce code breakage, we're using
+  -- `forall proxy. proxy sym` here so that `SProxy` code will still compile.
+  -- When PureScript makes a new breaking release after the v0.14.0 release,
+  -- this type signature will be updated to `Proxy sym -> String`.
+  reflectSymbol :: forall proxy. proxy sym -> String
 
 -- local definition for use in `reifySymbol`
 foreign import unsafeCoerce :: forall a b. a -> b
 
-reifySymbol :: forall r. String -> (forall sym. IsSymbol sym => SProxy sym -> r) -> r
-reifySymbol s f = coerce f { reflectSymbol: \_ -> s } SProxy where
+reifySymbol :: forall r. String -> (forall sym. IsSymbol sym => Proxy sym -> r) -> r
```

Comment:
I think this should use a type variable `proxy` instead of `Proxy`, for the same reason as `reflectProxy` - that is, so that it works with both Proxy and SProxy until we update it again later.

Reply:
Good catch! I've fixed that in latest commits.